controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -38,14 +44,27 @@
 /**
  * Class to encapsulate the logic to report Controller service metrics for Streams.
  */
+@Slf4j
 public final class StreamMetrics extends AbstractControllerMetrics implements AutoCloseable {
 
+    private final int statsOutputFrequencySeconds;
+
     private final OpStatsLogger createStreamLatency = STATS_LOGGER.createStats(CREATE_STREAM_LATENCY);
     private final OpStatsLogger deleteStreamLatency = STATS_LOGGER.createStats(DELETE_STREAM_LATENCY);
     private final OpStatsLogger sealStreamLatency = STATS_LOGGER.createStats(SEAL_STREAM_LATENCY);
     private final OpStatsLogger updateStreamLatency = STATS_LOGGER.createStats(UPDATE_STREAM_LATENCY);
     private final OpStatsLogger truncateStreamLatency = STATS_LOGGER.createStats(TRUNCATE_STREAM_LATENCY);
 
+    private ScheduledExecutorService executor = Executors.newScheduledThreadPool(2);
```

Comment:
I don't see a reason to introduce another thread pool. More fundamentally, I don't see a reason for having a reporting period. The problem I have observed is that values reported are not monotonically increasing, which I'm not sure why it is happening.

Reply:
In addition to what @fpj said:
1. This thread pool is not created using `ExecutorServiceHelpers`. The factory methods in there take care of a number of other things for you.
2. This is not a final variable

I recommend you reuse an existing threadpool and pass it via the constructor. That way we don't have idle threads sitting around for most of the time.